File: app/src/main/java/com/haishinkit/util/Log.java
package com.haishinkit.util;

public final class Log {
    private static ILogAdapter adapter = new SystemLogAdapter();

    static {
        String runtime = System.getProperty("java.runtime.name");
        if (0 <= runtime.indexOf("Android")) {
            adapter = new AndroidLogAdapter();
        }
    }

    public static int v(String tag, String msg) {
        return adapter.v(tag, msg);
    }

    public static int v(String tag, String msg, Throwable tr) {
        return adapter.v(tag, msg, tr);
    }

    public static int d(String tag, String msg) {
        return adapter.v(tag, msg);
    }

    public static int d(String tag, String msg, Throwable tr) {
        return adapter.d(tag, msg, tr);
    }

    public static int i(String tag, String msg) {
        return adapter.i(tag, msg);
    }

    public static int i(String tag, String msg, Throwable tr) {
        return adapter.i(tag, msg, tr);
    }

    public static int w(String tag, String msg) {
        return adapter.w(tag, msg);
    }

    public static int w(String tag, String msg, Throwable tr) {
        return adapter.w(tag, msg, tr);
    }

    public static int w(String tag, Throwable tr) {
        return adapter.w(tag, tr);
    }

    public static int e(String tag, String msg) {
        return adapter.e(tag, msg);
    }

    public static int e(String tag, String msg, Throwable tr) {
        return adapter.e(tag, msg, tr);
    }
}

File: haishinkit/src/main/java/com/haishinkit/util/Log.kt
package com.haishinkit.util

object Log {
    private var adapter: ILogAdapter = SystemLogAdapter()

    init {
        val runtime = System.getProperty("java.runtime.name")
        if (0 <= runtime.indexOf("Android")) {
            adapter = AndroidLogAdapter()
        }
    }

    fun v(tag: String, msg: String): Int {
        return adapter.v(tag, msg)
    }

    fun v(tag: String, msg: String, tr: Throwable): Int {
        return adapter.v(tag, msg, tr)
    }

    fun d(tag: String, msg: String): Int {
        return adapter.v(tag, msg)
    }

    fun d(tag: String, msg: String, tr: Throwable): Int {
        return adapter.d(tag, msg, tr)
    }

    fun i(tag: String, msg: String): Int {
        return adapter.i(tag, msg)
    }

    fun i(tag: String, msg: String, tr: Throwable): Int {
        return adapter.i(tag, msg, tr)
    }

    fun w(tag: String, msg: String): Int {
        return adapter.w(tag, msg)
    }

    fun w(tag: String, msg: String, tr: Throwable): Int {
        return adapter.w(tag, msg, tr)
    }

    fun w(tag: String, tr: Throwable): Int {
        return adapter.w(tag, tr)
    }

    fun e(tag: String, msg: String): Int {
        return adapter.e(tag, msg)
    }

    fun e(tag: String, msg: String, tr: Throwable): Int {
        return adapter.e(tag, msg, tr)
    }
}